### PR TITLE
reflect supported HW and OS in README

### LINKED ISSUE #846: State supported HW and OS
Clearly state in README what's supported and what's not.

We should also (technically) deny to install memory intensive extensions such as ZRAM or Grafana on boxes to have less than 1GB (e.g. RPi1, RPi0).

see #730 for a proposal
see https://github.com/orgs/openhab/teams/openhabian-maintainers/discussions/10

We would also need to state the OS we support see #621, #623 for OS

### PR DESCRIPTION
**Pull request by mstormi**
Supersedes #701 
Fixes #846 

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

**Comment by EliasGabrielsson** [comment]
We have not yet officially dropped support for pine64. 
Manual installation should still work. We can merge this when we are sure that the Armbian openhab installation script is on pair with openhabian.

**Comment by mstormi** [comment]
Hmm, spend a couple of more thoughts on this please.

* today we don't have ANY (clear) definition of what is supported and what is not
  That unfortunately results in people trying to use all sorts of ARM HW today, getting frustrated after spending hours on unsuccessful installations and to then blame openHABian for not working properly. That's what I want to avoid.
Some of these are edge cases it's really hard to argument against (e.g. I learned that there is a box called CM3 made by Raspberry foundation, designed for industrial use, but it does not work with current openHABian).

* we haven't been testing on pine64 HW for a long time, right ? So as per my suggested definition of "supported", which includes to test any code change against real HW, pine64 has not been supported in a long time.
On the other hand, if you please read my proposal again, my definition of "unsupported" does not imply it doesn't work. It does not mean we prohibit installations there or remove code that is required to run on such HW. It just means we don't test on this HW. 
I think it is crucial to have a proper starting point.
Unfortunately it's unknown how long that Armbian stuff will take to be completed so we should not wait for that to happen first to avoid frustration going on.
I therefore suggest to add the "supported HW" section asap. We can change it again anytime later.
If you dislike my definition then propose your changes to the wording using code review comments.

File: README.md [hunk 1]
```diff
@@ -48,21 +74,27 @@ Testing is done continuously with Travis using the test framework [Bats](https:/
 
 ```
 docker build --tag openhabian/openhabian-bats .
-docker run -it openhabian/openhabian-bats bash -c 'bats -r -f "unit-." .'
+docker run -i openhabian/openhabian-bats bash -c 'bats -r -f "unit-." .'
 
 docker run --name "install-test" --privileged -d openhabian/openhabian-bats
+<<<<<<< HEAD
```
Comment (holgerfriedrich) [inline README.md:80]: merge conflict?

**Comment by mstormi** [comment]
@openhab/openhabian-maintainers please let me know you opinion on this.
Do you share my position on (all) RPi and Ubuntu to be supportworthy ?

File: README.md (hunk 1 above)
Comment (mstormi) [reply README.md]: yes, resolved

**Comment by mstormi** [comment]
> @openhab/openhabian-maintainers please let me know your opinion on this.

@openhab/openhabian-maintainers @holgerfriedrich @ecdye please get me your feedback on the 
README draft in this PR by this weekend latest. 

=> Is _that_ what we want to support in terms of HW and OS from now on ?

I'd really like to get a statement on that out the door to avoid more false user expectations.
Will merge this if I don't get to see a veto from you.

PS: note this PR is just the *docs* - no code changes yet

File: README.md [hunk 2]
```diff
@@ -12,21 +12,45 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
 The project provides two things:
 
-*   A set of scripts to set up openHAB on any Debian/Ubuntu based system
-*   Complete **SD-card images pre-configured with openHAB** and many other openHAB- and Hardware-specific preparations, for the *Raspberry Pi* and the *Pine A64* platforms.
+*   a set of scripts to set up openHAB on any Debian based system (Raspbian, Ubuntu)
+*   a complete **SD-card image pre-configured with openHAB** and many other openHAB- and Hardware-specific preparations for all *Raspberry Pi* models.
 
-Closely related to openHABian is the repository [openhab-linuxpkg](https://github.com/openhab/openhab-linuxpkg) which provides linux packages for openHAB. openHABian uses those packages to setup openHAB with some additional configurations.
+## Hardware and OS support
+As of openHABian version 1.5.1 of the image, all Raspberry Pi models are supported as hardware.
```
Comment (ecdye) [inline README.md:19]: No such image as of yet.

File: README.md [hunk 3]
```diff
@@ -12,21 +12,45 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+Anything x86 based may work or not. Anything else ARM based such as ODroids, OrangePis and the like may work or not.
+NAS servers such as QNAP and Synology boxes will not work. Support for PINEA64 was also dropped in this current release.
+We strongly recommend to choose Raspberry Pi 2, 3 or 4 systems to have 1GB of RAM or more. RPi 1 and 0/0W only have a single CPU core and 512MB. This can be sufficient to run a smallish openHAB setup, but it will not be enough to run a full-blown system with many bindings and memory consuming openHABian features/components such as ZRAM, InfluxDB or Grafana.
```
Comment (ecdye) [inline README.md:22]: Should be "We strongly recommend that users choose ..."

File: README.md [hunk 4]
```diff
@@ -12,21 +12,45 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+We do not actively prohibit installation on any hardware - including unsupported systems -, but we might skip or deny to install specific extensions such as those memory hungry features named above.
+Supporting a hardware means to test every single patch and every release. There's simply too many SBC and HW combinations we don't have available or the time to spend on testing efforts it takes to make openHABian a reliable system.
```
Comment (ecdye) [inline README.md:24]: Better said as "Supporting hardware means testing every single patch and every release. There are simply too many SBC and HW combinations that maintainers do not have available, or, even if they did, the time to spend on testing efforts that are required to make openHABian a reliable system."

File: README.md [hunk 5]
```diff
@@ -12,21 +12,45 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+We recommend to use Ubuntu for x86 hardware and ARMbian for ARM hardware that we don't support.
+Note we expect you to use an up to date distribution: 'Buster' for Raspbian and 'Bionic Beaver' for Ubuntu, these are. To install openHABian on anything older may work or not.
+
+Either way, please note that you're on your own when it comes to configuring and installing the HW with the proper OS yourself.
+
+### 64 bit ?
+Although RPi3 and 4 have a 64 bit processor, you cannot run openHAB in 64 bit.
+The Azul Java Virtual Machine we currently use is incompatible with the aarch64 ARM architecture.
+In general you should be aware that to run in 64 bit has a major drawback, doubled memory usage that essentially is.
+Not quite a good idea on a heavily memory constrained platform like a RPi.
```
Comment (ecdye) [inline README.md:38]: Combine these statements "In general you should be aware that to run in 64-bits has a major drawback, doubled memory usage, that is not a good idea on a heavily memory constrained platform like a RPi.

File: README.md [hunk 6]
```diff
@@ -49,22 +73,22 @@ $ sudo bash build.bash platform dev-url branch url
...
-docker build --tag openhabian/openhabian-bats .
-docker run -it openhabian/openhabian-bats bash -c 'bats -r -f "unit-." .'
+docker build -t openhabian/openhabian-bats .
+docker run -i openhabian/openhabian-bats bash -c 'bats -r -f "unit-." .'
 
 docker run --name "install-test" --privileged -d openhabian/openhabian-bats
-docker exec -it install-test bash -c "./build.bash local-test && mv ~/.profile ~/.bash_profile && /etc/rc.local"
-docker exec -it install-test bash -c 'bats -r -f "installation-." .'
-docker exec -it install-test bash -c 'bats -r -f "destructive-." .'
+docker exec -i install-test bash -c "./build.bash local-test && mv ~/.profile ~/.bash_profil && /etc/rc.local"
```
Comment (ecdye) [inline README.md:80]: Get rid of the extraneous spaces, and shouldn't `~/.bash_profil` be `~/.bash_profile`

File: docs/openhabian-DEBUG.md [hunk 7]
```diff
@@ -81,9 +82,12 @@ Remember to always let `openhabian-config` update itself on start.
 
 If you want to change anything to work around some not yet fixed bug, you can directly edit the files in and below `/opt/openhabian` on your box. Just do not let openhabian-config update itself on start as that would overwrite your changes.
 
+You can also clone (download) a different openHABian version than the most current one, e.g. if someone offers or asks you to testdrive a development version.
```
Comment (ecdye) [inline docs/openhabian-DEBUG.md:85]: Probably better said as "if a maintainer / contributor to openHABian offers or asks you to testdrive a development version."

File: functions/menu.bash [hunk 8]
```diff
@@ -19,7 +30,7 @@ show_main_menu() {
   "00 | About openHABian    "    "Information about the openHABian project and this tool" \
   "" "" \
   "01 | Update"                  "Fetch the latest revision of the openHABian Configuration Tool" \
-  "02 | Upgrade System"          "Upgrade all installed software packages to their newest version" \
+  "02 | Upgrade System"          "Upgrade all installed software packages (incl. openHAB2) to their latest version" \
```
Comment (ecdye) [inline functions/menu.bash:33]: Probably isn't necessary to state that, as it should be assumed to be included by stating all installed software packages.

File: README.md (hunk 2 above)
Comment (mstormi) [reply README.md:19]: done

File: README.md (hunk 3 above)
Comment (mstormi) [reply README.md:22]: done

File: README.md (hunk 4 above)
Comment (mstormi) [reply README.md:24]: ... that IS required..., no?

File: README.md (hunk 6 above)
Comment (mstormi) [reply README.md:80]: what extraneous spaces?

File: functions/menu.bash (hunk 8 above)
Comment (mstormi) [reply functions/menu.bash:33]: So I thought, too. But there was criticism, some users didn't understand that. So I think it doesn't do harm to explicitly state that.

File: docs/openhabian-DEBUG.md (hunk 7 above)
Comment (mstormi) [reply docs/openhabian-DEBUG.md]: ok

File: README.md (hunk 5 above)
Comment (mstormi) [reply README.md]: ok

File: README.md (hunk 6 above)
Comment (ecdye) [reply README.md]: perhaps it was just a bug with github, it looked like there were a bunch of extra spaces on the end of line 80

File: README.md (hunk 4 above)
Comment (ecdye) [reply README.md]: actually "... the time to spend on the testing efforts that is required ..." but yes.

File: functions/menu.bash (hunk 8 above)
Comment (ecdye) [reply functions/menu.bash:33]: Ok

File: README.md [hunk 9]
```diff
@@ -12,21 +12,44 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+Supporting hardware means testing every single patch and every release. There are simply too many SBC and HW combinations that maintainers do not have available, or, even if they did, the time to spend on the testing efforts that is required to make openHABian a reliable system.
+Let's make sure you understand the implications of these statements: it means that to run hardware other than RPi 2/3/4 or x86 is **not** _fully_ supported.
+You can try and there's a good chance it'll work out for you. But if you hit any problem related to memory or hardware, you'll be on your own. You are expected **not** to raise these problems as issues on the community forum or on GitHub.
 
-## Installation and Setup
+Going beyond what the RPi image provides, as a manually installed set of scripts, openHABian is supposed to run on all Debian like Linux distributions on either ARM or x86 hardware. Note our automated testing for x86 is based on generic Debian and we take no resposibilities if you run into trouble with a different Linux flavour.
+We recommend to use Ubuntu for x86 hardware and ARMbian for ARM hardware that we don't support.
```
Comment (holgerfriedrich) [inline README.md:29]: I cannot recommend Ubuntu. I am a Debian user. New Ubuntu release always bring the more recent software versions, which gave us some headaches the last weeks (Python3, etc.)
My recommendation would be clearly Debian.

File: README.md [hunk 10]
```diff
@@ -12,21 +12,44 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+Note we expect you to use an up to date distribution: 'Buster' for Raspbian and 'Bionic Beaver' for Ubuntu, these are. To install openHABian on anything older may work or not.
```
Comment (holgerfriedrich) [inline README.md:30]: next Ubuntu LTS version was 20.04 was released (Focal Fossa), but I don't think that we are fully compatible yet.
My recommendation would be not to mention Ubuntu at all...

File: README.md [hunk 11]
```diff
@@ -12,21 +12,44 @@ Setting up a fully working Linux system with all needed packages and openHAB rec
...
+
+Either way, please note that you're on your own when it comes to configuring and installing the HW with the proper OS yourself.
+
+### 64 bit ?
+Although RPi3 and 4 have a 64 bit processor, you cannot run openHAB in 64 bit.
+The Azul Java Virtual Machine we currently use is incompatible with the aarch64 ARM architecture.
+In general you should be aware that to run in 64 bit has a major drawback, doubled memory usage, that is not a good idea on a heavily memory constrained platform like a RPi.
```
Comment (holgerfriedrich) [inline README.md:37]: still correct?
btw: I would not day "doubled mem", better "increased mem"

File: functions/menu.bash [hunk 12]
```diff
@@ -2,11 +2,22 @@
 
 show_about() {
   whiptail --title "About openHABian and openhabian-config" --msgbox "openHABian Configuration Tool $(get_git_revision)
-  \\nThis tool provides a few routines to make your openHAB experience as comfortable as possible. The menu options help with the setup and configuration of your system. Please select a menu entry to learn more.
-  \\nVisit the following websites for more information:
+This tool provides a little help to make your openHAB experience as comfortable as possible.
+\\nMake sure you have read the README and know about the Debug and Amanda Backup guides in /opt/openhabian/docs.
```
Comment (holgerfriedrich) [inline functions/menu.bash:6]: I would not explicitly name Amanda here, just backup would be good (do not confuse newbies)

File: functions/menu.bash (hunk 8 above)
Comment (holgerfriedrich) [inline functions/menu.bash:33]: remove 2, it will be outdated at some time :-)

File: README.md (hunk 10 above)
Comment (mstormi) [reply README.md:30]: Ok, I've reworded that.
But to destroy hopes, I'm afraid that will not relieve us from a need to support Ubuntu ...

Focal is latest, I believe it is the equivalent of debian testing/bullseye, isn't it ? or that of unstable/sid ?
Did Ubuntu move from bionic to disco (dingo) as the "stable" equivalent of debian buster when they released focal ?

File: README.md (hunk 9 above)
Comment (mstormi) [reply README.md:29]: I understand you, I feel the same.
But the challenge to maintainership is we need to support what people use, not what we like best. Remember Travis testing is also based on Ubuntu, not Raspbian.

File: functions/menu.bash (hunk 8 above)
Comment (mstormi) [reply functions/menu.bash:33]: ok good point :-)

File: functions/menu.bash (hunk 12 above)
Comment (mstormi) [reply functions/menu.bash:6]: ok

File: README.md (hunk 11 above)
Comment (mstormi) [reply README.md:37]: ok yes other than that it's still correct

File: README.md (hunk 9 above)
Comment (mstormi) [reply README.md]: I changed that to make clear Ubuntu is not supported for the time being.

I think we will need to support it in the future, though. So many more to use it than Debian.

Comment (ecdye) [reply README.md]: I agree, I think it will be a large benefit to us if we offer Ubuntu support in addition to Debian as Ubuntu is much more widely used.

**Comment by mstormi** [comment]
FYI [skip ci] in commit messages will make Travis skip builds. No need to burn CPU cycles when we just change docs.

File: README.md (hunk 10 above)
Comment (holgerfriedrich) [reply README.md]: Seems to be finally released
https://wiki.ubuntu.com/FocalFossa/ReleaseNotes

Comment (mstormi) [reply README.md]: Yes, I read and caught up on Ubuntu release cycle now.
Focal is the new current long term support version, Bionic was the previous one.
We should take care of both of these (anyone to know a good docker image source for it ?),
Will create an issue to track introduction of support.

But as written: for now, Ubuntu is out of scope.

Thanks for the review.
This is the first time ever for openHABian to make a statement on supported HW and OS.
I think this was overdue. Let's see how it does in terms of reducing false expectations.

**Comment by ecdye** [comment]
> FYI [skip ci] in commit messages will make Travis skip builds. No need to burn CPU cycles when we just change docs.

Good to know, I've been wondering how to do that.

**Comment by mstormi** [comment]
> FYI [skip ci] in commit messages will make Travis skip builds.

There's also an option in Travis settings to cancel running builds when you upload an update.

**Pull request merged**